tests/4.5/target_teams_distribute/test_target_teams_distribute_nowait.F90

```diff
@@ -0,0 +1,79 @@
+!===--- test_target_teams_distribute_nowait.F90-----------------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the nowait clause on a target teams distribute directive and
+! uses a barrier to resyncronize the target regions.  Since we can't be sure
+! that operations will be asyncronous, we can not test to make sure that
+! the regions are executed asynchronously.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+#define N_TASKS 16
+
+PROGRAM test_target_teams_distribute_nowait
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+  INTEGER :: errors
+  OMPVV_TEST_OFFLOADING
+  errors = 0
+
+  OMPVV_WARNING("This test does not throw an error if tasks fail to execute asynchronously, as this is still correct behavior. If execution is not asynchronous, we will throw a warning.")
+
+  OMPVV_TEST_VERBOSE(test_nowait() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+CONTAINS
+  INTEGER FUNCTION test_nowait()
+    INTEGER:: errors, x, y, z, was_async, my_ticket
+    INTEGER,DIMENSION(N_TASKS):: order
+    INTEGER,DIMENSION(N_TASKS, N):: work_storage
+    INTEGER,DIMENSION(1):: ticket
+
+    ticket(1) = 0
+    errors = 0
+    was_async = 0
+
+    !$omp target enter data map(to: ticket(1:1), order(1:N_TASKS), my_ticket)
+
+    DO x = 1, N_TASKS
+       !$omp target teams distribute map(alloc: work_storage(x, 1:N), ticket(1:1)) private(my_ticket) nowait
```

Comment:
xlf won't compile because of non-contiguous memory here, I believe it relates to the work_storage mapping in particular.